consensus/hotstuff/eventhandler/event_handler.go

```diff
@@ -247,11 +338,55 @@ func (e *EventHandler) startNewView() error {
 	if e.committee.Self() == currentLeader {
 		log.Debug().Msg("generating block proposal as leader")
 
+		// TODO(active-pacemaker): SafetyRules checks identity using proposal.BlockID, here we can check only by view ?
+		// check if we are eligible to propose
+		_, err = e.committee.IdentityByEpoch(curView, e.committee.Self())
+		if err != nil {
+			if model.IsInvalidSignerError(err) {
+				// we are ejected at this epoch
```

Comment:
In my mind, this check is redundant with https://github.com/onflow/flow-go/blob/542c0928f9f7f44264c4527688ec005c1568c495/consensus/hotstuff/eventhandler/event_handler.go#L325-L328
I think we can entirely remove it.

regarding your TODO: 
>  SafetyRules checks identity using proposal.BlockID, here we can check only by view ?

* The difference is that this logic is executed only if _this_ node is the proposer. Generally, we write the logic for honest nodes, i.e. we don't consider the case in our implementation where _this_ node is byzantine. Hence, without loss of generality we assume that if I am elected primary for a particular view that I am not ejected 
* Even if I am ejected, I can still propose and the other nodes have to handle this. Self-ejection of honest nodes is rare enough that we can accept the little bit of overhead. Presumably, a node operator would take their node down anyway. And the nodes are selected as primary rarely enough that the overhead of adding a proposal is negligible. 
* In contrast, the safety rules implementation has to deal with _other_ nodes being byzantine. So there, is is a likely case that an ejected byzantine node is still trying to propose. 

I know that the reasoning is a bit subtle, but hopefully it makes sense (?) 😅

Reply:
I agree that this check is redundant with `LeaderForView` since both of them operate on full epoch data, what I would like to avoid is an ejected replica(by any reason) trying to propose. I was thinking that we could raise a slashing challenge if this  happens, but the question is do we even plan on doing this? If we don't plan to slash this type of behavior then I agree that we could just ignore proposals of ejected nodes by not voting for them(that's what we have right now in `SafetyRules`). 
To summarize:
- if we don't plan to slash ejected nodes trying to propose then we should remove this check and rely on `SafetyRules` on not voting. 
- If we do plan to slash, then we should add some kind of check for primary to omit creating proposal that can be potentially slashed. That's the purpose of TODO that I have left. I think we should use `IdentityByBlock` instead of `IdentityByEpoch` but should discuss it after answering main question.